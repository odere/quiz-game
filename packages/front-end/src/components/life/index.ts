import Life from './life.view';

export default Life
