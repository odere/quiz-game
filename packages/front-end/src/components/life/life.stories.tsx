import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';

import Life from './life.view';


const Template: ComponentStory<typeof Life> = (args: any) => <Life {...args} />;

export const Primary = Template.bind({});

Primary.args = {
    hasLife: true,
};

export const NoLife = Template.bind({});

NoLife.args = {
    hasLife: false,
};

export default {
    title: 'Components/Life',
    component: Life,
} as ComponentMeta<typeof Life>
