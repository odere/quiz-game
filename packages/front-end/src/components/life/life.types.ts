export interface LifeStyledProps {
	hasLife: boolean
}

export interface LifeProps extends LifeStyledProps {
	onClick(): void
	label: string
}
