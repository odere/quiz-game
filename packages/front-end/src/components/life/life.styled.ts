import styled from 'styled-components'

import { LifeStyledProps } from './life.types'

import { green, primary, red } from '../../constants/colors.constants'

export const StyledLife = styled.div`
	background: ${(props: LifeStyledProps) => (props.hasLife ? green : red)};
	color: ${primary};
	font-weight: 600;
	line-height: 1.5em;
	padding: 0.25em;
	text-align: center;
	transition: 200ms;
	width: 50%;

	&:hover {
		filter: brightness(1.25);
	}
`
