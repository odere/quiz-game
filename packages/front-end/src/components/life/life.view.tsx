import React from 'react'

import { StyledLife } from './life.styled'
import { LifeProps } from './life.types'

const Life: React.FC<LifeProps> = props => {
	const { hasLife = true, onClick, label } = props

	return (
		<StyledLife hasLife={hasLife}>
			<div onClick={onClick}>{label}</div>
		</StyledLife>
	)
}

export default Life
