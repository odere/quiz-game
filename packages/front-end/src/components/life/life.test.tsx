import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import { StyledLife } from './life.styled'

describe('components/Life:', () => {
	it('should be able to change theme', () => {
		const treePrimary = renderer.create(<StyledLife hasLife />).toJSON()
		expect(treePrimary).toMatchSnapshot()

		const treeSecondary = renderer.create(<StyledLife hasLife={false} />).toJSON()
		expect(treeSecondary).toMatchSnapshot()
	})
})
