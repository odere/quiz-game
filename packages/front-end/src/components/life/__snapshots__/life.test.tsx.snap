// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Life: should be able to change theme 1`] = `
.c0 {
  background: #66bd38;
  color: #44168b;
  font-weight: 600;
  line-height: 1.5em;
  padding: 0.25em;
  text-align: center;
  -webkit-transition: 200ms;
  transition: 200ms;
  width: 50%;
}

.c0:hover {
  -webkit-filter: brightness(1.25);
  filter: brightness(1.25);
}

<div
  className="c0"
/>
`;

exports[`components/Life: should be able to change theme 2`] = `
.c0 {
  background: #ff3859;
  color: #44168b;
  font-weight: 600;
  line-height: 1.5em;
  padding: 0.25em;
  text-align: center;
  -webkit-transition: 200ms;
  transition: 200ms;
  width: 50%;
}

.c0:hover {
  -webkit-filter: brightness(1.25);
  filter: brightness(1.25);
}

<div
  className="c0"
/>
`;
